include/aspect/material_model/grain_size.h

```diff
@@ -192,6 +197,9 @@ namespace aspect
         std::vector<double> reciprocal_required_strain;
         std::vector<double> recrystallized_grain_size;
 
+        SUNDIALS::ARKode<VectorType>::AdditionalData data;
+        mutable VectorType grain_sizes;
```

Comment:
Do these two variables need to be class members? I think you could probably just move those into the function where you end up using those, and that makes the function thread-safe too.

Reply:
I first had this in the function, but @gassmoeller suggested to move them out to make it faster (so we do not have to create them again for every cell).